lib/phoenix/socket/transport.ex

```diff
@@ -273,8 +273,10 @@ defmodule Phoenix.Socket.Transport do
     :noreply
   end
 
-  defp log_info("phoenix" <> _, _func), do: :noop
-  defp log_info(_topic, func), do: Logger.info(func)
+  defp log(_, "phoenix" <> _, _func), do: :noop
+  defp log(%{ private: %{ log_join: false } }, _topic, _func), do: :noop
+  defp log(%{ private: %{ log_join: level } }, _topic, func), do: Logger.log(level, func)
+  defp log(_, _topic, func), do: Logger.info(func)
```

Comment:
It should probably just be killed. I couldn't find any tests which exercised the log-flag functionality, so I kept it to ensure backwards compatibility in case I had missed a detail in the code which could mean that a FunctionClauseError would occur.